FBDeviceControl/Utility/FBDeviceControlFrameworkLoader.m

```diff
@@ -164,4 +156,40 @@ + (void)enableDVTDebugLogging
   [[objc_lookUpClass("DVTLogAspect") logAspectWithName:@"CommandInvocation"] setLogLevel:10];
 }
 
++ (BOOL)isAtLeastMacOSSierra
+{
+  NSOperatingSystemVersion sierra = {10, 12, 0};
+  return [[NSProcessInfo processInfo] isOperatingSystemAtLeastVersion:sierra];
+}
+
++ (BOOL)isAtLeastXcode81
+{
+  NSDecimalNumber *xcodeVersion = FBControlCoreGlobalConfiguration.xcodeVersionNumber;
+  NSDecimalNumber *xcode81 = [NSDecimalNumber decimalNumberWithString:@"8.1"];
+  return [xcodeVersion compare:xcode81] != NSOrderedAscending;
+}
+
++ (NSArray<FBWeakFramework *> *)privateFrameworks
+{
+  NSArray<FBWeakFramework *> *frameworks =
+  @[
+    FBWeakFramework.DTXConnectionServices,
+    FBWeakFramework.DVTFoundation,
+    FBWeakFramework.IDEFoundation,
+    FBWeakFramework.IDEiOSSupportCore,
+    FBWeakFramework.IBAutolayoutFoundation,
+    FBWeakFramework.IDEKit,
+    FBWeakFramework.IDESourceEditor
+    ];
+
+  if (FBDeviceControlFrameworkLoader.isAtLeastMacOSSierra &&
```

Comment:
Here would be a suitable place for adding a description of what differences there are in the Frameworks across macOS/Xcode versions. This will help for future reference.